pinot-core/src/main/java/org/apache/pinot/core/util/validator/PinotConfigValidator.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.util.validator;
+
+import com.github.fge.jsonschema.core.exceptions.ProcessingException;
+import java.io.Serializable;
+import java.util.List;
+
+
+public interface PinotConfigValidator extends Serializable {
+
+  boolean validate(String jsonStr)
+      throws ProcessingException;
+
+  List<String> getValidationMessages();
```

Comment:
I wonder if we should have just one method - that takes a jsonStr and returns a list of validation messages. That way there's no dependency on the caller side to invoke validate first and then invoke getValidationMessages.